CDEFramework/Libraries/LegacyDictionarySupport/Configuration/WebAPISection.cs

```diff
@@ -0,0 +1,71 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Configuration;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace LegacyDictionarySupport.Configuration
+{
+    /// <summary>
+    /// Configuration information for a Web API
+    /// </summary>
+    public class WebAPISection : ConfigurationSection
+    {
+        private static readonly string CONFIG_SECTION_NAME = "legacyServices/webAPI";
+
+        /// <summary>
+        /// Gets the host name of the Web API server.
+        /// </summary>
+        [ConfigurationProperty("apiHost", IsRequired=true)]
+        public string APIHost
+        {
+            get { return (string)base["apiHost"]; }
+        }
+
+        /// <summary>
+        /// Gets the port number for the API
+        /// </summary>
+        [ConfigurationProperty("apiPort", IsRequired=false)]
+        public string APIPort
+        {
+            get { return (string)base["apiPort"]; }
+        }
+
+        /// <summary>
+        /// Gets the protocol for the API
+        /// </summary>
+        [ConfigurationProperty("apiProtocol", IsRequired = true)]
+        public string APIProtocol
+        {
+            get { return (string)base["apiProtocol"]; }
+        }
+
+        /// <summary>
+        /// Gets the URL for the Web API from the configuration
+        /// </summary>
+        public static string GetAPIUrl()
+        {
+            string url = "";
+            WebAPISection config = (WebAPISection)ConfigurationManager.GetSection(CONFIG_SECTION_NAME);
+
+            if (config == null)
+                throw new ConfigurationErrorsException("The configuration section, " + CONFIG_SECTION_NAME + ", cannot be found");
+
+            if (string.IsNullOrWhiteSpace(config.APIProtocol))
+                throw new ConfigurationErrorsException(CONFIG_SECTION_NAME + "error: apiProtocol cannot be null or empty");
+
+            if (string.IsNullOrWhiteSpace(config.APIHost))
+                throw new ConfigurationErrorsException(CONFIG_SECTION_NAME + "error: apiHost cannot be null or empty");
+
+            url = string.Format("{0}://{1}", config.APIProtocol, config.APIHost);
+
+            if (!string.IsNullOrWhiteSpace(config.APIPort))
+            {
+                url += ":" + config.APIPort;
+            }
+
+            return url;
+        }
```

Comment:
This is how we set up the GetAPIUrl for clinical trials search and the web API used by best bets. @bryanpizzillo wanted these to be separate and configurable in the case that any of these changes. Unless he chimes in that it doesn't matter and can be changed to all one method...

Reply:
I may be mistaken, but I think I’m the one who originated the pattern of breaking out the protocol, host, and port, and @bryanpizzillo complained it should have all been one piece.

I don’t recall whether I did that for CTS (maybe not, I worked on “Clippy” for the front end), but I do recall that being the old pattern and we’ve since moved to combining them.